ArcadiaParties-SPA/src/app/user/selector/selector.ts

```diff
@@ -0,0 +1,15 @@
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+
+import { UserState, userFeatureStateKey } from '../reducers/user-reducer';
+
+export const userFeatureSelector = createFeatureSelector<UserState>(userFeatureStateKey);
+
+export const selectUserIdentity = createSelector(
+    userFeatureSelector,
+    state => state.user.identity
```

Comment:
You can use nested selectors here. Instead of using feature selector, you can use `selectUser` selector. Also, please add check if user is not defined.

Reply:
Now using nested selectors